applications/tari_base_node/src/builder.rs

```diff
@@ -227,7 +227,7 @@ pub fn configure_and_initialize_node(
         DatabaseType::Memory => {
             let rules = ConsensusManager::default();
             let backend = MemoryDatabase::<HashDigest>::default();
-            let mut db = BlockchainDatabase::new(backend).map_err(|e| e.to_string())?;
+            let mut db = BlockchainDatabase::new(backend, &rules).map_err(|e| e.to_string())?;
```

Comment:
consensus manager is already an arc, we can just clone and move in a copy, we dont need to send in a ref

Reply:
I think we've had this debate before. The callee can decide to clone, and even those clones are cheap, references are even cheaper

Reply:
I'm leaving it with a reference for now as the blockchain db doesn't require permanent access to the consensus rules at this point.